fbw-a380x/src/systems/instruments/src/FCU/FcuFsInstrument.tsx

```diff
@@ -0,0 +1,151 @@
+   */
+  constructor(public readonly instrument: BaseInstrument) {
+    // force enable animations
+    document.documentElement.classList.add('animationsEnabled');
+
+    this.renderComponents();
+
+    this.backplane.addInstrument('Clock', this.clock);
+    this.backplane.addPublisher('HEvent', this.hEventPublisher);
+
+    this.backplane.addInstrument('AltitudeManager', new AltitudeManager(this.bus));
+    this.backplane.addInstrument('AutopilotManager', new AutopilotManager(this.bus));
+    this.backplane.addInstrument('BaroManager1', new BaroManager(this.bus, 1), true);
+    this.backplane.addInstrument('HeadingManager', new HeadingManager(this.bus));
+    this.backplane.addInstrument('SpeedManager', new SpeedManager(this.bus));
+    this.backplane.addInstrument('VerticalSpeedManager', new VerticalSpeedManager(this.bus));
+
+    this.backplane.addPublisher('FcuPublisher', new FcuPublisher(this.bus));
+    this.backplane.addPublisher('FgBusPublisher', new FGDataPublisher(this.bus));
+    this.backplane.addPublisher('MsfsAutopilotAssistancePublisher', new MsfsAutopilotAssitancePublisher(this.bus));
+    this.backplane.addPublisher('MsfsRadioNavigationPublisher', new MsfsRadioNavigationPublisher(this.bus));
+
+    this.doInit();
+  }
+
+  /** @inheritdoc */
+  private renderComponents(): void {
+    FSComponent.render(
+      <FcuDisplay bus={this.bus} isHidden={this.isOperating.map(SubscribableMapFunctions.not())} />,
+      document.getElementById('FCU_CONTENT'),
+    );
+
+    // Remove "instrument didn't load" text
+    document.getElementById('FCU_CONTENT')?.querySelector(':scope > h1')?.remove();
+  }
+
+  /** @inheritdoc */
+  public Update(): void {
+    this.failuresConsumer.update();
+    // this.isFailed.set(this.failuresConsumer.isActive(this.isFailedKey));;
+
+    this.backplane.onUpdate();
+  }
+
+  /** @inheritdoc */
+  public onInteractionEvent(args: string[]): void {
+    this.hEventPublisher.dispatchHEvent(args[0]);
+  }
+
+  /** @inheritdoc */
+  public onFlightStart(): void {
+    // noop and not useful
+  }
+
+  /** @inheritdoc */
+  public onGameStateChanged(oldState: GameState, newState: GameState): void {
+    // noop
+  }
+
+  /** @inheritdoc */
+  public onSoundEnd(soundEventId: Name_Z): void {
+    // noop
+  }
+
+  public onPowerOn(): void {
+    this.isPowered.set(true);
+  }
+
+  public onPowerOff(): void {
+    this.isPowered.set(false);
+  }
+
+  /** Init instrument. */
+  private doInit(): void {
+    // this.failuresConsumer.register(this.isFailedKey);
+
+    // very dubious code from old FCU
+    this.isOperating.sub((v) => {
+      if (v) {
+        if (!SimVar.GetSimVarValue('AUTOPILOT FLIGHT DIRECTOR ACTIVE:1', 'bool')) {
+          SimVar.SetSimVarValue('K:TOGGLE_FLIGHT_DIRECTOR', 'number', 1);
+        }
+        if (!SimVar.GetSimVarValue('AUTOPILOT FLIGHT DIRECTOR ACTIVE:2', 'bool')) {
+          SimVar.SetSimVarValue('K:TOGGLE_FLIGHT_DIRECTOR', 'number', 2);
+        }
+      }
+    });
```

Comment:
This is pretty dubious, maybe add a TODO or FIXME to the comment so we know to try and refactor this out if possible one day.